account_invoice_inter_company/models/account_move.py

```diff
@@ -214,6 +233,29 @@ def button_cancel(self):
                     inter_invoice.button_cancel()
         return super().button_cancel()
 
+    def write(self, vals):
+        res = super().write(vals)
+        for move in self.filtered(
+            lambda x: x.sudo().auto_invoice_id and x.state == "posted"
```

Comment:
Remove the posted check. In fact, you can only modify an invoice in draft state.